Frontend/ui-driver/src/Screens/HomeScreen/Controller.purs

```diff
@@ -390,7 +392,87 @@ eval (RideActionModalAction (RideActionModal.MessageCustomer)) state = do
     _ <- pure $ setValueToLocalStore LOCAL_STAGE (show ST.ChatWithCustomer)
     _ <- pure $ setValueToLocalNativeStore READ_MESSAGES (show (Array.length state.data.messages))
-    continue state{props{currentStage = ST.ChatWithCustomer, sendMessageActive = false, unReadMessages = false, isChatOpened = true}}
+    let allMessages = getChatMessages ""
+    continue state{data{messages = allMessages}, props{currentStage = ST.ChatWithCustomer, sendMessageActive = false, unReadMessages = false, isChatOpened = true}}
+
+eval (UpdateInChat) state = continue state {props{updatedArrivalInChat = true}}
+
+eval (InitializeChat ) state = continue state {props{chatcallbackInitiated = true}}
+
+eval RemoveChat state = do
+  continueWithCmd state {props{chatcallbackInitiated = false}} [ do
+    _ <- stopChatListenerService
+    _ <- pure $ setValueToLocalNativeStore READ_MESSAGES "0"
+    pure $ NoAction
+  ]
+
+eval (UpdateMessages message sender timeStamp size) state = do
+  if not state.props.chatcallbackInitiated then continue state {props {canSendSuggestion = true}} else do
+    continueWithCmd state{data{messagesSize = size}, props {canSendSuggestion = true}} [do
+      pure $ (RideActionModalAction (RideActionModal.LoadMessages))
+    ]
+    
+eval (RideActionModalAction (RideActionModal.LoadMessages)) state = do
+  let allMessages = getChatMessages ""
+  case (Array.last allMessages) of
+      Just value -> if value.message == "" then continue state {data { messagesSize = show (fromMaybe 0 (fromString state.data.messagesSize) + 1)}, props {canSendSuggestion = true}} else
+                      if value.sentBy == "Driver" then updateMessagesWithCmd state {data {messages = allMessages, suggestionsList = []}, props {canSendSuggestion = true}}
+                      else do
+                        let readMessages = fromMaybe 0 (fromString (getValueToLocalNativeStore READ_MESSAGES))
+                        let unReadMessages = (if (readMessages == 0 && state.props.currentStage /= ST.ChatWithCustomer) then true else (if (readMessages < (Array.length allMessages) && state.props.currentStage /= ST.ChatWithCustomer) then true else false))
+                        let suggestions = getSuggestionsfromKey value.message
+                        updateMessagesWithCmd state {data {messages = allMessages, suggestionsList = suggestions }, props {unReadMessages = unReadMessages, canSendSuggestion = true}}
+      Nothing -> continue state {props {canSendSuggestion = true}}
+
+eval ScrollToBottom state = do
+  _ <- pure $ scrollToEnd (getNewIDWithTag "ChatScrollView") true
+  continue state
+
+eval (ChatViewActionController (ChatView.TextChanged value)) state = do
+  let sendMessageActive = if (length (trim value)) >= 1 then
+                          true
+                        else
+                          false
+  continue state{data{messageToBeSent = (trim value)},props{sendMessageActive = sendMessageActive}}
```

Comment:
you can simplify this with one line `length (trim value) >= 1` which will return bool